starlette/requests.py

```diff
@@ -251,10 +252,10 @@ async def close(self) -> None:
 
     async def is_disconnected(self) -> bool:
         if not self._is_disconnected:
-            try:
-                message = await asyncio.wait_for(self._receive(), timeout=0.0000001)
-            except asyncio.TimeoutError:
-                message = {}
+            message: Message = {}
+            with anyio.move_on_after(0.001):
+                # XXX: to small of a deadline and this blocks
```

Comment:
Sounds brittle?

Reply:
@uSpike what do you mean by blocks? - are we getting a negative deadline?

Reply:
That was discovered early in my development and it may have changed with anyio3 or other changes.  What I found was that  a deadline lower than around 0.001 (such as 0.00001) would cause the tests  to block.  I'll investigate more.

Reply:
I was able to refactor this into an event with 4dd8c5de79df2ee56959dba7bc20bbcdbb4d833d

Reply:
It seems I resolved the wrong conversation, my bad.

Reply:
I'm not quite sure what to do here.  The original code seems brittle to me as well:
```python
try:
    message = await asyncio.wait_for(self._receive(), timeout=0.0000001)
except asyncio.TimeoutError:
    message = {}
```
What is the original intention here?

Reply:
The commend `# XXX: to small of a deadline and this blocks` is incorrect.  Too small of a value for `anyio.move_on_after` and it seems that `self._receive()` doesn't have enough time to execute and `self._disconnected` is not set, and the tests fail.

Reply:
@JayH5 @tomchristie  OK, this is (actually) fixed now.  @graingert helped me to understand that the test client `async def receive()` needs to be non-trapping (meaning it doesn't `await`) when `response_complete` is already set.  This allows a zero deadline in `is_disconnected` which is much cleaner.

Reply:
Amazing, thanks